common-app-lib/src/cmr/common_app/api/launchpad_token_validation.clj

```diff
@@ -1,17 +1,66 @@
 (ns cmr.common-app.api.launchpad-token-validation
   "Validate Launchpad token."
   (:require
+   [cheshire.core :as json]
+   [clojure.data.codec.base64 :as base64]
+   [clojure.string :as string]
    [cmr.common-app.config :as config]
    [cmr.common.services.errors :as errors]
    [cmr.transmit.config :as transmit-config]))
 
+;; Note: Similar code exists at gov.nasa.echo.kernel.service.authentication
 (def URS_TOKEN_MAX_LENGTH 100)
 
+(defn is-heritage-token?
+  "Heritage token, meaning not the original style legacy token, but a modern
+   token made to be identifiable but also comply with the legacy standard. These
+   tokens are prefixed with 'EDL+'.
+   Note: Similar code exists at gov.nasa.echo.kernel.service.authentication."
+  [token]
+  (string/starts-with? token "EDL+"))
+
+(defn is-legacy-token?
+  "A check of last resort, only used if all other checks have not passed.
+   Note: Similar code exists at gov.nasa.echo.kernel.service.authentication."
+  [token]
+  (and (not (is-heritage-token? token))
+       (<= (count token) URS_TOKEN_MAX_LENGTH)))
+
+(defn is-jwt-token?
```

Comment:
use `defn-` for private functions. This also apply to other functions defined above.